frappe/integrations/doctype/dropbox_settings/dropbox_settings.py

```diff
@@ -152,19 +152,27 @@ def upload_file_to_dropbox(filename, folder, dropbox_client):
 	f = open(encode(filename), 'rb')
 	path = "{0}/{1}".format(folder, os.path.basename(filename))
 
-	if file_size <= chunk_size:
-		dropbox_client.files_upload(f.read(), path, mode)
-	else:
-		upload_session_start_result = dropbox_client.files_upload_session_start(f.read(chunk_size))
-		cursor = dropbox.files.UploadSessionCursor(session_id=upload_session_start_result.session_id, offset=f.tell())
-		commit = dropbox.files.CommitInfo(path=path, mode=mode)
-
-		while f.tell() < file_size:
-			if ((file_size - f.tell()) <= chunk_size):
-				dropbox_client.files_upload_session_finish(f.read(chunk_size), cursor, commit)
-			else:
-				dropbox_client.files_upload_session_append(f.read(chunk_size), cursor.session_id,cursor.offset)
-				cursor.offset = f.tell()
+	try:
+		if file_size <= chunk_size:
+			dropbox_client.files_upload(f.read(), path, mode)
+		else:
+			upload_session_start_result = dropbox_client.files_upload_session_start(f.read(chunk_size))
+			cursor = dropbox.files.UploadSessionCursor(session_id=upload_session_start_result.session_id, offset=f.tell())
+			commit = dropbox.files.CommitInfo(path=path, mode=mode)
+
+			while f.tell() < file_size:
+				if ((file_size - f.tell()) <= chunk_size):
+					dropbox_client.files_upload_session_finish(f.read(chunk_size), cursor, commit)
+				else:
+					dropbox_client.files_upload_session_append(f.read(chunk_size), cursor.session_id,cursor.offset)
+					cursor.offset = f.tell()
+	except dropbox.exceptions.ApiError as e:
+		if isinstance(e.error, dropbox.files.UploadError):
```

Comment:
you will have to import `dropbox.files` first, no?

Also why on directly `except dropbox.files.UloadError:` ?

Reply:
- you will have to import dropbox.files first, no?
 as we have imported dropbox package we can use directly 

- Also why on directly except dropbox.files.UloadError:
As per dropbox docstring, in case of the exception, the method raises UploadError.